warehouse/oidc/models/google.py

```diff
@@ -0,0 +1,98 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from sqlalchemy import Column, ForeignKey, String, UniqueConstraint
+from sqlalchemy.dialects.postgresql import UUID
+
+from warehouse.oidc.models._core import (
+    OIDCPublisher,
+    PendingOIDCPublisher,
+    _check_claim_binary,
+    _check_claim_invariant,
+)
+
+
+class GooglePublisherMixin:
+    """
+    Common functionality for both pending and concrete Google OIDC
+    providers.
+    """
+
+    email = Column(String, nullable=False)
+    sub = Column(String, nullable=False)
+
+    __required_verifiable_claims__ = {
+        "email": _check_claim_binary(str.__eq__),
+        "email_verified": _check_claim_invariant(True),
```

Comment:
Flagging for review: `email_verified` should always be `true`, so this just enforces that.